src/dspbase.jl

```diff
@@ -207,17 +207,48 @@ conv2(A::StridedMatrix{T}, B::StridedMatrix{T}) where {T<:Integer} =
 conv2(u::StridedVector{T}, v::StridedVector{T}, A::StridedMatrix{T}) where {T<:Integer} =
     round.(Int, conv2(float(u), float(v), float(A)))
 
+function check_padmode_kwarg(padmode::Symbol, su::Integer, sv::Integer)
+    if padmode == :default_longest
+        if su != sv
+            Base.depwarn(
+            """
+            The default behavior for xcorr will changed in future versions.
+            For more details, see the documentation for xcorr. To avoid this
+            warning, specify padmode = :none or padmode = :longest where
+            appropriate
+            """
+                ,
+                :xcorr
+            )
+        end
+        :longest
+    else
+        padmode
+    end
+end
+
 """
-    xcorr(u,v)
+    xcorr(u,v; padmode = :longest)
 
-Compute the cross-correlation of two vectors.
+Compute the cross-correlation of two vectors. The size of the output depends on
+`padmode` keyword argument: if `padmode = :none` then the result will be the same size
+as [`conv`](@ref) of `u` and `v`. If `padmode = :longest` then the result will have
+length `2*max(length(X), length(Y))-1`, where the beginning of the result will
+be padded with zeros. If `padmode = :none`, it will be `length(X) - length(Y) - 1`.
```

Comment:
The pre-padding or post-padding of the result depend on which of the arguments ends up getting padded (you can confirm in the tests below). So if the time-reversed argument (`v`) gets padded then the result is pre-padded, but if `u` gets padded then the result is post-padded.

also an editing nitpick, but I'd put the two clauses relating to `:none` together, so something like:

> Compute the cross-correlation of two vectors. The size of the output depends on
the `padmode` keyword argument: with `padmode = :none` the length of the
result will be  `length(u) + length(v) - 1`, as with [`conv`](@ref). With
`padmode = :longest` the shorter of the arguments will be padded so they
are equal length. This gives a result with length `2*max(length(u), length(v))-1`,
with the zero-lag condition at the center.

We should also add a note about the deprecation and upcoming change:
```
!!! warning
    The default value of `padmode` will be changing from `:longest` to `:none`
    in a future release of DSP. In preparation for this change, leaving `padmode` unspecified
    is currently deprecated.
```

Reply:
Whoops, didn't re-read the doc string before addressing your comment. Good catch!